### DISCUSSION TITLE
Руководство ПЛИ, тело документа, п.1.5 Секция "Результаты проведенных исследований" RESLAB

### DISCUSSION THREAD

**Discussion by eugenekogan-spb** [fhir-ru/core#284] [Лабораторная РГ]
У3-4
Обязательная секция.
Секция предназначена для указания результатов проведенных лабораторных исследований и организована в виде иерархической структуры:
• Корневой элемент, объединяющий все проведенные в рамках данного протокола лабораторные исследования с возможностью дать общее, интегральное лабораторное заключение по всему документу;
• Уровень группировок исследований, объединяющий лабораторные показатели в структуры, имеющие общее происхождение (полученные из одного образца или с одного анализатора) или требующие совместную интерпретацию (например, проба Реберга для оценки клиренса креатинина из крови в мочу). При необходимости к такой группировке может быть дано текстовое примечание. На этом же уровне возможно прикрепление клинических вложений в документ, например: графиков, фотографий микропрепаратов, гистологических сканов и т.п.
• Уровень лабораторных показателей – атомарных результатов лабораторного исследования, детализированных на необходимом и достаточном для сравнения между собой уровне.

**Comment by eugenekogan-spb** [comment]
(не привожу скриншот секции в CDA, поскольку он очень длинный)

**Reply by eugenekogan-spb** [reply]
**здесь можно привести логическую схему информации в секции**

**Comment by eugenekogan-spb** [comment]
**Причина отмены исследования**

1. МОЖЕТ иметь один [0..1] элемент code (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.1063", атрибут @codeSystemName должен иметь значение "Причины отмены выполнения лабораторного исследования", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий причину отмены выполнения лабораторного исследования. Возможны следующие альтернативы:
- ЕСЛИ атрибут statusCode/@code имеет значение "cancelled", ТО элемент обязан быть представлен;
- ЕСЛИ атрибут statusCode/@code имеет значение "completed", ТО элемент должен отсутствовать;

**Reply by eugenekogan-spb** [reply]
Видимо, следует предусмотреть расширение для DiagnosticReport

```
* extension[+]
  * url = "diagnosticreport-cancel-reason"
  * valueCodeableConcept
    * text = "Нарушение правил взятия материала"
    * coding
      * system = "1.2.643.5.1.13.13.99.2.1063"
      * code = #1
```

**Reply by maximalexeev** [reply]
Предлагаю использовать причины, предусмотренные в Task и Observation, DR оставить без детализации - тем более что связь ДР и О чаще всего 1:1

**Reply by eugenekogan-spb** [reply]
О.Пензин: если для причины невыполнения исследования делать расширение, то семантически естественно делать его на уровне DiagnosticReport, поскольку именно этот ресурс отвечает за содержание исследования, в то время как Composition отвечает за документ. Не "документ не выполнен", а "исследование не выполнено".
М.Алексеев: надо на уровне Composition, потому что если все исследование не выполнено, то причина должна быть указана один раз, в то время как DiagnosticReport несколько.

С другой стороны, возможно, следует в этом случае (только для не выполненного исследования) включить в комплект ресурс Task и отразить причину невыполнения в Task.statusReason

Решили обратиться к эксперту @Vrodvi

**Reply by Vrodvi** [reply]
Composition - это документ, просто группировка информации, в нём не может содержаться информации по сутевой части.

Task - это шаг некого процесса, внутренности, стоящие за тем как из поступившего направления на исследование сделать само исследование и составить результаты. Скорее всего в документы эти внутренности попадать не должны.
Но если чисто технически говорить, то именно в этом ресурсе есть готовый статус "не выполнено" и есть атрибут для описания причины. Можно притянуть за уши и сказать, что на каждое направление нужно формировать один формальный Task и в нём фиксировать статус выполнения всего исследования в целом, но этого как будто использование не совсем по предназначению.

> потому что если все исследование не выполнено, то причина должна быть указана один раз, в то время как DiagnosticReport несколько.

Ну и почему DiagnosticReport должно быть несколько? Это как раз ресурс для формирования итогового цельного отчёта по исследованию, если несколько отчётов, то и исследований видимо несколько? В описании ресурса явно говорится, что он может описывать как простые исслдеования, так и комплексные - см. https://hl7.org/fhir/diagnosticreport.html#10.3.6.4 

**Про само "исследование не выполнено" - почему оно не выполнено, на каком этапе? В зависимости от ответа может понадобиться кодирование в разных местах.**
1) Если было направление, но лаборатория решила вообще его не брать на исполнение (или сам заказчик отменил) - тогда это https://hl7.org/fhir/servicerequest.html и в нём статус [revoked](https://hl7.org/fhir/codesystem-request-status.html#request-status-revoked).
2) Если направление принято, но не выполнено исследование какой-то части исследования - тогда это в https://hl7.org/fhir/observation.html указание статуса [cancelled](https://hl7.org/fhir/codesystem-observation-status.html#observation-status-cancelled), есть прям готовый пример в точности как в вашем случае text = "Нарушение правил взятия материала" - https://hl7.org/fhir/observation-example-unsat.html
3) Если все части исследования выполнены, но результата каким-то образом недостаточно для того, чтобы исследование в целом считать "выполненным" - тогда это уже DiagnosticReport, т.е. в итоговом заключении будет каким-то образом сказано, что исследование не выполнено, хотя все тесты проведены и даже будут указаны их резульататы.

**Reply by eugenekogan-spb** [reply]
Список возможных причин невыполнения исследования в НСИ МЗ РФ - https://nsi.rosminzdrav.ru/dictionaries/1.2.643.5.1.13.13.99.2.1063/passport/1.1 - нарушение правил взятия, хранения, транспортировки или первичной обработки материала. Из этого следует, что имеется в виду п.1 по списку В.Родионова - лаборатория решила не брать заказ на исполнение. 
При этом размещение информации о причинах неисполнения в ресурсе направления выглядит нелогично: это не свойство направления, это результат оценки материала, оценку выполнила лаборатория, то есть - информация принадлежит DiagnosticReport. Получается, выживает предложение о добавлении расширения в DiagnosticReport.

**Comment by eugenekogan-spb** [comment]
**статус выполнения**

2. ОБЯЗАН иметь один [1..1] элемент statusCode (атрибут @code должен быть представлен с не пустым значением). Возможны следующие альтернативы:
- ЕСЛИ лабораторное исследование выполнено, ТО атрибут statusCode/@code должен иметь значение "completed";
- ЕСЛИ лабораторное исследование не выполнено, ТО атрибут statusCode/@code должен иметь значение "cancelled";

**Reply by eugenekogan-spb** [reply]
Информация соответствует атрибуту DiagnosticRerort.status, имеющему тип данных code и привязку required. Требование использования только двух из возможных значений атрибута можно реализовать маппингом набора значений Diagnostic Report Status (registered | partial | preliminary | modified | final | amended | corrected | appended | cancelled | entered-in-error | unknown) на два значения (completed и cancelled).

```
* status = #final
```

Комментарий: получается, что передача частичных и предварительных результатов лабораторных исследований не поддержана текущей спецификацией СЭМД.

**Comment by eugenekogan-spb** [comment]
**профиль по справочнику профилей ФСЛИ**

1. ДОЛЖЕН иметь один [1..1] элемент code (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1437", атрибут @codeSystemName должен иметь значение "Федеральный справочник лабораторных исследований. Профили лабораторных исследований", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий указание произвольной группировки исследований. ЕСЛИ значение не может быть закодировано с использованием справочника "1.2.643.5.1.13.13.11.1437", ТО атрибут @nullFlavor должен иметь значение "OTH". Данный элемент:
1. МОЖЕТ иметь один [0..1] элемент originalText, который должен иметь не пустое наполнение. ЕСЛИ атрибут code/@nullFlavor имеет значение "OTH", ТО элемент обязан быть представлен и быть заполнен значением "Отдельные лабораторные тесты";

**Reply by eugenekogan-spb** [reply]
Требуется расширение DiagnosticReport.

```
* extension[+]
  * url = "predefinedTestGroup"
  * valueCodeableConcept
    * text = "Общий анализ крови (сокращенный)"
    * coding
      * system = "1.2.643.5.1.13.13.11.1437"
      * code = #10023
```

**Comment by eugenekogan-spb** [comment]
**выполненные услуги**

3. МОЖЕТ иметь произвольное количество [0..*] элементов component, содержащих сведения о выполненной медицинской услуге, который:
3.1. ОБЯЗАН иметь один [1..1] элемент observation (атрибут @classCode должен иметь значение "OBS", атрибут @moodCode должен иметь значение "EVN"), который:
3.1.1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "833", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.166", атрибут @codeSystemName должен иметь значение "Кодируемые поля CDA документов", атрибут @displayName должен иметь значение "Медицинская услуга", атрибут @codeSystemVersion должен быть представлен с не пустым значением), содержащий код поля;
3.1.2. ОБЯЗАН иметь один [1..1] элемент value (атрибут @xsi:type должен иметь значение "CD ", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1070", атрибут @codeSystemName должен иметь значение "Номенклатура медицинских услуг", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий код медицинской услуги;

**Reply by eugenekogan-spb** [reply]
Можно предположить использование DiagnosticReport.supportingInfo, однако этот элемент предназначен для передачи информации, использованной при выполнении исследования - "This backbone element contains supporting information that was used in the creation of the report not included in the results already included in the report." Услуги никак не используются при выполнении исследования, поэтому представляется более корректным использовать расширение.

```
* extension[+]
  * url = "procedureCode"
  * valueCodeableConcept
    * text = "Общий (клинический) анализ крови"
    * coding
      * system = "1.2.643.5.1.13.13.11.1070"
      * code = #B03.016.002
```

**Reply by maximalexeev** [reply]
А почему не DiagnosticReport.code?

**Reply by eugenekogan-spb** [reply]
Разобрали на встрече 12.11.24 (тайм-код 42:01). Оставили extension, потому что code возможен только один, а услуг, связанных с DR, может быть несколько.

Принято решение, что группировать данные в протоколе по DR следует по профилям исследований, то есть по происхождению выполненных тестов.

**Comment by eugenekogan-spb** [comment]
**сведения об исследовании лабораторного показателя: код, статус, время**

1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1080", атрибут @codeSystemName должен иметь значение "Федеральный справочник лабораторных исследований. Справочник лабораторных тестов", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий код лабораторного теста;
2. ОБЯЗАН иметь один [1..1] элемент statusCode (атрибут @code должен быть представлен с не пустым значением). Возможны следующие альтернативы:
- ЕСЛИ лабораторный тест выполнен, ТО атрибут statusCode/@code должен иметь значение "completed";
- ЕСЛИ лабораторный тест не выполнен, ТО атрибут statusCode/@code должен иметь значение "cancelled";
3. МОЖЕТ иметь один [0..1] элемент effectiveTime (атрибут @value должен быть представлен с не пустым значением), содержащий дату и время выполнения исследования. Возможны следующие альтернативы:
- ЕСЛИ атрибут statusCode/@code имеет значение "cancelled", ТО элемент может отсутствовать;
- ЕСЛИ атрибут statusCode/@code имеет значение "completed", ТО один элемент обязан быть представлен;

**Reply by eugenekogan-spb** [reply]
Лабораторные показатели передаются в виде набора ресурсов Observation
Код без особенностей.
Статус реализуется атрибутом Observation.status, имеющему тип данных code и привязку required. Поэтому, аналогично статусу всего DiagnosticReport, можно обеспечить передачу двух требуемых спецификацией значений путем маппинга кодов.
Дата-время без различий между FHIR и CDA. Но важно отметить, что это клинически значимое время результата (обычно время забора материала).

```
Instance: diagnostic-report-pli
InstanceOf: DiagnosticReport

* result = Reference (observation-gematocrit)

Instance: observation-gematocrit
InstanceOf: Observation

* code
  * text = "Гематокрит крови методом автоматизированного подсчёта"
  * coding
    * system = "1.2.643.5.1.13.13.11.1080"
    * version = "3.40"
    * code = #1017359

* status = #completed

* effectiveDateTime = "2022-11-15T08:50:00+0400"
```

**Comment by eugenekogan-spb** [comment]
**сведения об исследовании лабораторного показателя: значение**

4.МОЖЕТ иметь один [0..1] элемент value (атрибут @xsi:type должен быть представлен с не пустым значением), содержащий результат исследования лабораторного показателя. Возможны следующие альтернативы:
4.1. ЕСЛИ атрибут statusCode/@code имеет значение "cancelled", ТО атрибут value/@xsi:type должен иметь значение "CD", атрибут value/@codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.1063", атрибут value/@codeSystemName должен иметь значение "Причины отмены выполнения лабораторного исследования", атрибуты value/@code, value/@codeSystemVersion, value/@displayName должны быть представлены с не пустыми значениями;
4.2. ЕСЛИ атрибут statusCode/@code имеет значение "completed", ТО один элемент должен быть представлен; Возможны следующие альтернативы:
4.2.1. ЕСЛИ атрибут value/@xsi:type имеет значение "ST", ТО элемент value должен иметь не пустое наполнение;
4.2.2. ЕСЛИ атрибут value/@xsi:type имеет значение "BL", ТО атрибут value/@value должен иметь значение "true" или "false";
4.2.3. ЕСЛИ атрибут value/@xsi:type имеет значение "PQ", ТО атрибут value/@value должен быть представлен со значением измеренного параметра, атрибут value/@unit должен быть представлен со значением из справочника "Единицы измерения" (OID: 1.2.643.5.1.13.13.11.1358), поле "Код UCUM";
4.2.4. ЕСЛИ атрибут value/@xsi:type имеет значение "CD", ТО:

4.2.4.1.ЕСЛИ кодируется информация о группе крови, ТО атрибут value/@codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1061", атрибут value/@codeSystemName должен иметь значение "Группы крови для учета сигнальной информации о пациенте", атрибут value/@code должен быть представлен с не пустым значением, атрибут value/@codeSystemVersion должен быть представлен с не пустым значением, атрибут value/@displayName должен быть представлен с не пустым значением;
4.2.4.2. ЕСЛИ кодируется информация о бактериях, ТО атрибут
value/@codeSystem должен значение "1.2.643.5.1.13.13.11.1087", value/@codeSystemName должен значение "Федеральный справочник исследований. Справочник бактерий", атрибут value/@code должен быть представлен с не пустым значением, атрибут value/@codeSystemVersion должен быть представлен с не пустым значением, атрибут value/@displayName должен быть представлен с не пустым значением;
4.2.4.3.ЕСЛИ кодируется информация о грибах, ТО атрибут
value/@codeSystem должен
значение "1.2.643.5.1.13.13.11.1088", value/@codeSystemName должен
значение Федеральный справочник
исследований. Справочник грибов", атрибут value/@code должен быть представлен с не пустым значением, атрибут value/@codeSystemVersion должен быть представлен с не пустым значением, атрибут value/@displayName должен быть представлен с не пустым значением;
4.2.4.4. ЕСЛИ иначе, ТО атрибуты value/@code, value/@codeSystem, value/@codeSystemName, value/@codeSystemVersion, value/@displayName должны быть представлены с не пустыми значениями;

4.2.5. ЕСЛИ атрибут value/@xsi:type имеет значение "IVL_PQ", ТО возможны альтернативы использования элементов value/low и value/high, описанные ниже;

**Reply by eugenekogan-spb** [reply]
4.1. Для передачи информации о не выполненном тесте в составе выполненного исследования спецификация предлагает передать тест, но в поле результата указать код по справочнику причин отмены.
В FHIR для этой цели можно передать такую информацию как причину отсутствия значения. Observation.dataAbsentReason имеет привязку Extensible к набору значений, в котором есть позиция not-performed. Можно добавить на втором уровне уточняющие позиции, почему именно он not-performed. 

Комментарий: В справочнике МЗ РФ причин неисполнения теста перечислены причины, связанные с преаналитикой. Они применимы прежде всего к отмене исследования в целом. Для отмены конкретного теста могут быть и иные причины, например, связанные с самой лабораторией.

```
Instance: observation-gematocrit
InstanceOf: Observation

* code
  * text = "Гематокрит крови методом автоматизированного подсчёта"
  * coding
    * system = "1.2.643.5.1.13.13.11.1080"
    * version = "3.40"
    * code = #1017359

* status = #cancelled

* dataAbsentReason
  * text = "Нарушение правил взятия материала"
  * coding
    * system = "1.2.643.5.1.13.13.99.2.1063"
    * code = #1

```

4.2. Для передачи значений групп крови, грибов и бактерий будут использоваться соответствующие справочники. Можно будет сделать профили Observation для этих вариантов тестов, и в профиле ограничить использование только одним справочником.

```
Instance: observation-blood-group
InstanceOf: Observation

* code
  * text = "Группа крови"
  * coding
    * system = "1.2.643.5.1.13.13.11.1080"
    * version = "3.40"
    * code = #1038611

* status = #final

* valueCodeableConcept
  * text = "O (I)"
  * coding
    * system = "1.2.643.5.1.13.13.11.1061"
    * code = #101

Instance: observation-fungi
InstanceOf: Observation

* code
  * text = "Грибковый микроорганизм, определение культуральным методом"
  * coding
    * system = "1.2.643.5.1.13.13.11.1080"
    * version = "3.40"
    * code = #1132950

* status = #final

* valueCodeableConcept
  * text = "Genus Acremoniella"
  * coding
    * system = "1.2.643.5.1.13.13.11.1088"
    * code = #6000202

Instance: observation-bacteria
InstanceOf: Observation

* code
  * text = "Бактерии в моче качественно"
  * coding
    * system = "1.2.643.5.1.13.13.11.1080"
    * version = "3.40"
    * code = #1020312

* status = #final

* valueCodeableConcept
  * text = "Laribacter hongkongensis"
  * coding
    * system = " 1.2.643.5.1.13.13.11.1087"
    * code = #5027123
```

**Comment by eugenekogan-spb** [comment]
**единица измерения и референтный интервал**

3.МОЖЕТ иметь один [0..1] элемент translation (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1358", атрибут @codeSystemName должен иметь значение "Единицы измерения", атрибуты @value, @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение измеряемого параметра). Возможны следующие альтернативы:
1. ЕСЛИ атрибут value/@xsi:type имеет значение "PQ", ТО элемент обязан быть представлен;
2. ЕСЛИ иначе, ТО элемент должен отсутствовать;
4.МОЖЕТ иметь один [0..1] элемент low (атрибуты @value, @unit, @inclusive должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение измеряемого параметра, атрибут @unit должен иметь значение по справочнику "Единицы измерения" (OID: 1.2.643.5.1.13.13.11.1358), поле "Код UCUM"), содержащий нижняя граница интервала. Возможны следующие альтернативы:
1. ЕСЛИ атрибут value/@xsi:type имеет значение "IVL_PQ", ТО:
1. ЕСЛИ указанное значение включается в интервал, ТО атрибут low/@inclusive должен иметь значение "true";
2. ЕСЛИ иначе, ТО атрибут low/@inclusive должен иметь значение "false";
2. ЕСЛИ иначе, ТО элемент должен отсутствовать. Данный элемент low:
3.ОБЯЗАН иметь один [1..1] элемент translation (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1358", атрибут @codeSystemName должен иметь значение "Единицы измерения", атрибуты @value, @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение измеряемого параметра);
5. МОЖЕТ иметь один [0..1] элемент high (атрибуты @value, @unit, @inclusive должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение измеряемого параметра, атрибут @unit должен иметь значение по справочнику "Единицы измерения" (OID: 1.2.643.5.1.13.13.11.1358), поле "Код UCUM"), содержащий верхняя граница интервала. Возможны следующие альтернативы:
1. ЕСЛИ атрибут value/@xsi:type имеет значение "IVL_PQ", ТО:
1. ЕСЛИ указанное значение включается в интервал, ТО атрибут high/@inclusive должен иметь значение "true";
2. ЕСЛИ иначе, ТО атрибут high/@inclusive должен иметь значение "false";
2. ЕСЛИ иначе, ТО элемент должен отсутствовать. Данный элемент high:
3.ОБЯЗАН иметь один [1..1] элемент translation (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.11.1358", атрибут @codeSystemName должен иметь значение "Единицы измерения", атрибуты @value, @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями; атрибут @value должен иметь значение измеряемого параметра);

**Reply by eugenekogan-spb** [reply]
Единица измерения и референтный интервал - без особенностей.

```
* valueQuantity
  * value = 1500
  * unit = "мл"
  * system = "1.2.643.5.1.13.13.11.1358"
  * code = #16

* referenceRange
  * low
    * value = 1000
    * unit = "мл"
    * system = "1.2.643.5.1.13.13.11.1358"
    * code = #16
  * high
    * value = 2000
    * unit = "мл"
    * system = "1.2.643.5.1.13.13.11.1358"
    * code = #16
```

**Comment by eugenekogan-spb** [comment]
**интерпретация**

5. МОЖЕТ иметь один [0..1] элемент interpretationCode (атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.257", атрибут @codeSystemName должен иметь значение "Справочник кодов интерпретации результатов", атрибуты @code, @codeSystemVersion, @displayName должны быть представлены с не пустыми значениями), содержащий код интерпретации результата. Возможны следующие альтернативы:
- ЕСЛИ атрибут statusCode/@code имеет значение "cancelled", ТО элемент должен отсутствовать;
- ЕСЛИ атрибут statusCode/@code имеет значение "completed", ТО один элемент должен быть представлен;

**Reply by eugenekogan-spb** [reply]
Интерпретация передается атрибутом interpretation, который заполняется по справочнику http://hl7.org/fhir/valueset-observation-interpretation.html с привязной extensible.
Необходим маппинг справочника FHIR на справочник МЗРФ

```
* interpretation
  * text = "Нормальный (в пределах референсного диапазона)"
  * coding
    * system = "1.2.643.5.1.13.13.99.2.257"
    * code = #N
```

**Comment by eugenekogan-spb** [comment]
**образец**

6.МОЖЕТ иметь один [0..1] элемент specimen, содержащий сведения о материале или образце исследования. ЕСЛИ атрибут statusCode/@code имеет значение "cancelled", ТО элемент может отсутствовать; ЕСЛИ атрибут statusCode/@code имеет значение "completed", ТО один элемент обязан быть представлен. Данный элемент:
1. ОБЯЗАН иметь один [1..1] элемент specimenRole, который:
1. ОБЯЗАН иметь один [1..1] элемент id (атрибуты @root, @extension должны быть представлены с не пустыми значениями; атрибут @root должен быть сформирован по правилу формирования идентификаторов материалов для лабораторных исследований, т.е. "OID_медицинской_организации.100.НомерМИС.НомерЭкзМИ С.66", атрибут @extension должен соответствовать идентификатору материала для лабораторного исследования), содержащий уникальный идентификатор биоматериала или образца исследования в МИС (ссылку на секцию SPECIMENS);

**Reply by eugenekogan-spb** [reply]
Передается ссылкой на Specimen. В профиле Observation для ПЛИ необходимо ограничить ссылку только reference (Specimen) и не использовать Group.

```
* specimen = Reference (specimen-blood)
```

**Comment by eugenekogan-spb** [comment]
**исполнитель**

7.ОБЯЗАН иметь хотя бы один [1..*] элемент performer, содержащий сведения об исполнителе

**Reply by eugenekogan-spb** [reply]
Observation.performer, в нашем случае reference (PractitionerRole)

**Comment by eugenekogan-spb** [comment]
**оборудование и расходные материалы**

8. МОЖЕТ иметь произвольное количество элементов participant (атрибут @typeCode должен быть представлен с не пустым значением), содержащих сведения об использованном оборудовании или расходных материалах.

**Reply by eugenekogan-spb** [reply]
Вопрос: есть Observation.device, но он может быть только один

**Reply by eugenekogan-spb** [reply]
Видимо, здесь имеется в виду один прибор и пачка расходников.
расходник - возможно, extension с кодом по локальному справочнику.
Решили изучить R5.

**Comment by eugenekogan-spb** [comment]
9.МОЖЕТ иметь один [0..1] элемент entryRelationship (атрибут @typeCode должен иметь значение "COMP"), содержащий комментарий к лабораторному параметру, который:
... 2.ОБЯЗАН иметь один [1..1] элемент text, содержащий комментарий к лабораторному параметру, который должен иметь не пустое наполнение;
3.МОЖЕТ иметь один [0..1] элемент author, содержащий сведения об авторе комментария, который:
3.1. ОБЯЗАН иметь один [1..1] элемент time (атрибут @value должен быть представлен с не пустым значением), содержащий дату указания комментария;
3.2. ОБЯЗАН иметь один [1..1] элемент assignedAuthor (если медицинский работник присутствует в блоке заголовка "Сведения о документируемом событии" documentationOf, и у него имеется уникальный идентификатор в МИС, то допускается указание только уникального идентификатора автора в элементе assignedAuthor\id в качестве ссылки на его подробное описание в блоке documentationOf),...

**Reply by eugenekogan-spb** [reply]
Структуры идентичны. Информация передается в атрибуте note с типом данных Annotation, который включает author, time, text.

```
* note
  * authorReference = Reference (some-PractitionerRole)
  * time = "2022-11-15T08:50:00+0400"
  * text = "какой-то комментарий"
```

**Comment by eugenekogan-spb** [comment]
**заключение по исследованию с указанием автора и времени**

4. МОЖЕТ иметь один [0..1] элемент component, содержащий лабораторное заключение (в случае микробиологического исследования указывается заключение микробиологического исследования), который:
1. ОБЯЗАН иметь один [1..1] элемент act (атрибут @classCode должен иметь значение "ACT", атрибут @moodCode должен иметь значение "EVN"), который:
1. ОБЯЗАН иметь один [1..1] элемент code (атрибут @code должен иметь значение "901", атрибут @codeSystem должен иметь значение "1.2.643.5.1.13.13.99.2.166", атрибут @codeSystemName должен иметь значение "Кодируемые поля CDA документов", атрибут @displayName должен иметь значение "Текстовое заключение по проведенным лабораторным исследованиям", атрибут @codeSystemVersion должен быть представлен с не пустым значением), содержащий код поля;
2. ОБЯЗАН иметь один [1..1] элемент text, содержащий лабораторное заключение (текстом), который должен иметь не пустое наполнение;
3. МОЖЕТ иметь один [0..1] элемент author, содержащий сведения об авторе заключения, который:
1. ОБЯЗАН иметь один [1..1] элемент time (атрибут @value должен быть представлен с не пустым значением), содержащий дату указания заключения;
2.ОБЯЗАН иметь один [1..1] элемент assignedAuthor

**Reply by eugenekogan-spb** [reply]
Особенностей нет.
Информация может быть передана в атрибуте DiagnosticReport.note с типом данных Annotation, который включает author, time, text.

```
* note
  * authorReference = Reference (some-PractitionerRole)
  * time = "2022-11-15T08:50:00+0400"
  * text = "какой-то комментарий"
```

**Comment by eugenekogan-spb** [comment]
**Вопрос**
Референтный диапазон описан внутри элемента value (п.4.3, стр 95) и еще раз в информации о проведенном тесте (п.10 на стр 100). Это один и тот же диапазон?

**Comment by eugenekogan-spb** [comment]
**прикрепленные материалы**

5.МОЖЕТ иметь один [0..1] элемент component, содержащий прикрепленные данные по исследованию (например, микрофотографии или графики)

**Reply by eugenekogan-spb** [reply]
передается элементом DiagnosticReport.media

**Comment by eugenekogan-spb** [comment]
//в этом разделе приведем экземпляры ресурсов, необходимых для передачи информации из примера в СЭМД ПЛИ


```
Instance:   diagnostic-report-pli
InstanceOf: DiagnosticReport
Title: "Lab DiagnosticReport Instance PLI"
Description: "Результат лабораторного исследования от 31.10.2022"
Usage: #example

* status = #final

* category = http://terminology.hl7.org/CodeSystem/v2-0074#LAB

* code
  * text = "Какое-то лабораторное исследование"

* subject = Reference (patient-maria-antuanetta)

* encounter = Reference (encounter-outpatient)

* effectiveDateTime = "2022-10-31T21:50:00+04:00" 

* performer = Reference (practitionerrole-lab-doctor-helix-altay)

* basedOn
  * identifier
    * system = "1.2.643.5.1.13.13.12.2.22.1576.100.1.1.51"
    * value = "4632"
    * type
      * text = "Направление на микробиологическое исследование"
      * coding
        * system = "1.2.643.5.1.13.13.11.1522"
        * code = #331


//только для R5 - заполнена информация для секции витальных параметров
//* supportingInfo
//  * type
//    * text = "supporting clinical information"
//    * coding
//      * system = "http://terminology.hl7.org/CodeSystem/v2-0936"
//      * code = #SCI
//  * reference = Reference (observation-vital-sign-diuresis)

//далее информация для секции результатов

//статус выполнения
* status = #final

// профиль
* extension[+]
  * url = "predefinedTestGroup"
  * valueCodeableConcept
    * text = "Общий анализ крови (базовый)"
    * coding
      * system = "1.2.643.5.1.13.13.11.1437"
      * code = #10022

//медицинские услуги
* extension[+]
  * url = "procedureCode"
  * valueCodeableConcept
    * text = "Общий (клинический) анализ крови"
    * coding
      * system = "1.2.643.5.1.13.13.11.1070"
      * code = #B03.016.002

//результаты
* result[+] = Reference (observation-gematocrit)


Instance: observation-gematocrit
InstanceOf: Observation
Usage: #example

* code
  * text = "Гематокрит крови методом автоматизированного подсчёта"
  * coding
    * system = "1.2.643.5.1.13.13.11.1080"
    * version = "3.40"
    * code = #1017359

* status = #final

* effectiveDateTime = "2022-11-15T08:50:00+04:00"

* valueQuantity
  * value = 45.5
  * unit = "%"
  * system = "1.2.643.5.1.13.13.11.1358"
  * code = #53

* referenceRange
  * low
    * value = 39.0
    * unit = "%"
    * system = "1.2.643.5.1.13.13.11.1358"
    * code = #53
  * high
    * value = 49.0
    * unit = "%"
    * system = "1.2.643.5.1.13.13.11.1358"
    * code = #53

* interpretation
  * text = "Нормальный (в пределах референсного диапазона)"
  * coding
    * system = "1.2.643.5.1.13.13.99.2.257"
    * code = #N

* specimen = Reference (specimen-blood)

* performer = Reference (practitionerrole-lab-doctor-helix-altay)

* device = Reference (device-sysmex)

```
// для повышения читаемости примера опущены аналогичные результаты

// гемоглобин общий (код 1017128)
// 153 г/л (код 60), границы 132-173, N

// Эритроциты, количество в крови методом автоматизированного подсчёта (код 1017275)
// 5.38 unit="10^12/l" код 328, 4.30-5.70 N

// Средний объем эритроцита в крови методом автоматизированного подсчёта (код 1017441)
// 84.5" unit="fl" код 20, 80.0-99.0 N

**Reply by eugenekogan-spb** [reply]
Примечание: категория DR должна быть указана по справочнику МЗ РФ - см [документируемое событие](https://github.com/fhir-ru/core/discussions/278#discussioncomment-11230796)

**Comment by maximalexeev** [comment]
Е.Коган: Здесь был анонс микробиологии от Максима Алексеева, я перенес [в отдельную дискуссию](https://github.com/fhir-ru/core/discussions/288#discussion-7599714)

**Reply by eugenekogan-spb** [reply]
@OlegPenzin
